src/cmd/support_archive/tarball.go

```diff
@@ -0,0 +1,96 @@
+package support_archive
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"strings"
+	"time"
+)
+
+const tarFileName = "%s/operator-support-archive-%s.tgz"
+const defaultTargetDir = "/tmp/dynatrace-operator"
+
+type tarball struct {
+	tarWriter  *tar.Writer
+	gzipWriter *gzip.Writer
+	tarFile    *os.File
+}
+
+func newTarball(ctx *supportArchiveContext) (*tarball, error) {
```

Comment:
you are just using 2 things from the context, a bool and a string,

IMO this struct could be independent, and not be bound to the `supportArchiveContext`, so just pass in only the things that are necessary, nicer read (if not more then 3 params are used),nicer to test

Reply:
Agreed, although I find this 3-param rule a bit random. And in a way it prevents consistency of the function signatures.

Reply:
constructors and factory functions are exempt from that rule